ECO-8923: Create a contribution guide and add issue templates (#162)
fullstorydev/fullstory-babel-plugin-annotate-react: CONTRIBUTING.md

# Contributing to fullstory-babel-plugin-annotate-react

Thanks for taking the time to contribute!

Here are some important resources to consider:

  * [FullStory's Annotate React plugin for Web & Native](https://help.fullstory.com/hc/en-us/articles/360049493054-FullStory-s-Annotate-React-plugin-for-Web-Native)
  * [Getting Started with FullStory React Native Capture](https://help.fullstory.com/hc/en-us/articles/360052419133)

>   * [README.md](./README.md)

Reviewer: Nit: We could probably drop the file extension in the link text
```suggestion
  * [README](./README.md)
```

Author: fixed! and moved this to the top.

>   * [FullStory Help Center](https://www.fullstory.com/)

Reviewer: Should this link to `https://help.fullstory.com/` since the link text is for help center rather than the base FullStory website?

Author: fixed 👍  good idea.

  * [FullStory Community](https://community.fullstory.com/)
  * [FullStory Developer Guide](https://developer.fullstory.com/)

All types of contributions are encouraged and valued. See the [Table of Contents](#table-of-contents) for different ways to help.

## Table of Contents

- [I Have a Question](#i-have-a-question)
- [I Want To Contribute](#i-want-to-contribute)
    - [Making Code Changes](#making-code-changes)
    - [Reporting Bugs](#reporting-bugs)
    - [Suggesting Enhancements](#suggesting-enhancements)
- [Pull Request Process](#pull-request-process)

## I Have a Question

> > Search for similar questions in our [Community Portal](https://community.fullstory.com/) and browse through our [Help Center](https://help.fullstory.com/hc/en-us) before asking a question.

Reviewer: Should we link to the non `en-us` specific help center URL?
```suggestion
> Search for similar questions in our [Community Portal](https://help.fullstory.com/) and browse through our [Help Center](https://help.fullstory.com/hc/en-us) before asking a question.
```

Author: yes, but we want to keep the community center link it was suggested in ask-cx-support

Author: i also learned you can link to non-english articles like this:
```
# english
https://help.fullstory.com/hc/en-us/articles/360052419133

# without language specified
https://help.fullstory.com/hc/articles/360052419133
```

If you then still feel the need to ask a question and need clarification, we recommend:

> - Creating a new [GitHub issue](https://github.com/fullstorydev/fullstory-babel-plugin-annotate-react/issues/new?template=ask_a_question.md) with our ask a question template.

Reviewer: ```suggestion
- Creating a new [GitHub issue](https://github.com/fullstorydev/fullstory-babel-plugin-annotate-react/issues/new?template=ask_a_question.md), filling in the templated description.
```

Author: fixed 👍

	- Alternatively, you can submit your questions to our [Community](https://community.fullstory.com/).

> - Please provide as much context as you can about what you're running into.

Reviewer: ```suggestion
- Please provide as much context as you can with your question.
```

Author: fixed 👍

    1. **HTTPS**: `git clone https://github.com/fullstorydev/fullstory-babel-plugin-annotate-react.git`
    2. **SSH**: `git clone git@github.com:fullstorydev/fullstory-babel-plugin-annotate-react.git`
    3. **GitHub CLI**: `gh repo clone fullstorydev/fullstory-babel-plugin-annotate-react`
3. Install dependencies from the root directory of the project using `npm i`

#### Testing your changes

We use Jest as our test runner to verify changes and behaviors in our package. We use both unit tests and snapshot tests to prevent regressions in our code base. In order to run the jest tests you must have prepared your [local environment](#preparing-your-environment).

You can run the tests from the project root directory using `npm run test`.

### Reporting Bugs

#### Before Submitting a Bug Report

Please complete the following steps in advance to help us fix any potential bugs as fast as possible.

- Make sure that you are using the latest version.
- Determine if your bug is really a bug and not an error on your side e.g. using incompatible environment components/versions. (If you are looking for support, you might want to check [this section](#i-have-a-question)).

> - Check if other users have experienced (and potentially already solved) the same issue you are having, check if there is not already a bug report existing for your bug or error in the [bug tracker](https://github.com/fullstorydev/fullstory-babel-plugin-annotate-react/issues?q=is%3Aopen+is%3Aissue+label%3Abug).

Reviewer: ```suggestion
- Check if there is already an [open issue](https://github.com/fullstorydev/fullstory-babel-plugin-annotate-react/issues?q=is%3Aopen+is%3Aissue+label%3Abug) for the problem you're encountering.
```

Author: fixed 👍

- Make sure that you are using the latest version.
- Find out if the functionality is already covered.
- Find out whether your idea fits with the scope and aims of the project. It's up to you to convince the maintainers of the merits of this feature. We want features that will be useful to the majority of users.

#### Creating a Feature Request

Enhancements are suggested by creating a new [GitHub issue](https://github.com/fullstorydev/fullstory-babel-plugin-annotate-react/issues/new?&template=feature_request.md) with our feature request template.

- Use a descriptive title for the issue to identify the suggestion.
- Provide a step-by-step description of the suggested enhancement, include screenshots and animated GIFs if necessary.
- Describe the current behavior and explain which behavior you expected to see instead. 
- Describe alternatives to your solution and why the alternatives do not work for you.
- Explain why this enhancement would be useful.

## Pull Request Process

> All pull requests should include a clear list of your changes (read more about [pull requests](http://help.github.com/pull-requests/)). Any pull request that makes functional changes to the library should also include unit tests. We can always use more test coverage. 

Reviewer: The `pull requests` link seems to redirect to GitHub's base help page.

Author: yes, that was the intention; for beginners who might not be familiar with what pull request are. I can remove this section though 🤔 wdyt?

Reviewer: Maybe this would be a good link? https://docs.github.com/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests

Author: added that one 👍

Always write a clear log message for your commits. One-line messages are fine for small changes, but bigger changes should look like this:

```
$ git commit -m "A brief summary of the commit"
> 

> > A paragraph describing what changed and its impact."

Reviewer: Looks like there's a trailing `"` in the extended paragraph commit message.

Author: fixed should read like so. the user needs to enter a few new lines to add a description.

```
$ git commit -m "A brief summary of the commit
> 
> A paragraph describing what changed and its impact."
```
